common/appscale/common/async_retrying.py

```diff
@@ -157,39 +156,40 @@ def check_exception(error):
       else:
         check_exception = retry_on_exception
 
-      retries = 0
-      backoff = backoff_multiplier
-      start_time = time.time()
+      backoff_sequence = BackoffSequence(
+        base=backoff_base,
+        multiplier=backoff_multiplier,
+        threshold=backoff_threshold,
+        max_retries=max_retries,
+        timeout=retrying_timeout,
+        randomize=randomize_backoff
+      )
       node_lock = self._locks[node]
-      result = None
 
       # Wake older update calls (*)
       node_lock.condition.notify_all()
       node_lock.waiters += 1
       with (yield node_lock.lock.acquire()):
         node_lock.waiters -= 1
 
-        while True:
+        for backoff in backoff_sequence:
           # Start of retrying iteration
           try:
             result = func(*args, **kwargs)
             if isinstance(result, gen.Future):
               result = yield result
             break
 
+          except gen.Return:
```

Comment:
It looks like this path bypasses the lock release that happens at the end.

Reply:
Good point. And actually we don't need to catch it at all..
If it's coroutine, `gen.Return` exception is catch by tornado decorator, otherwise it shouldn't raise `gen.Return` at all, so we don't need to handle it differently comparing to any other Exception.

Thanks for pointing me to this mistake.